spec/system/projects/leader_searches_contributor_spec.rb

```diff
@@ -0,0 +1,160 @@
+require 'rails_helper'
+
+describe 'Líder de projeto pesquisa por colaboradores' do
+  it 'e deve estar autenticado' do
+    project = create :project
+
+    visit search_project_portifoliorrr_profiles_path project
+
+    expect(current_path).to eq new_user_session_path
+    expect(page).to have_content 'Para continuar, faça login ou registre-se'
+  end
+
+  context 'e não pode ser contribuidor' do
+    it 'para acessar a página' do
+      project = create :project
+      user = create :user, email: 'user@email.com', cpf: '149.759.780-32'
+      create :user_role, user:, project:, role: :contributor
+
+      login_as user
+      visit search_project_portifoliorrr_profiles_path project
+
+      expect(current_path).to eq root_path
+      expect(page).to have_content 'Você não tem acesso a esse recurso'
+    end
+
+    it 'para ver o botão' do
+      project = create :project
+      user = create :user, email: 'user@email.com', cpf: '149.759.780-32'
+      create :user_role, user:, project:, role: :contributor
+
+      login_as user
+      visit project_path project
+
+      expect(page).not_to have_link 'Procurar usuários'
+    end
+  end
+
+  context 'e não pode ser administrador' do
+    it 'para acessar a página' do
+      project = create :project
+      user = create :user, email: 'user@email.com', cpf: '149.759.780-32'
+      create :user_role, user:, project:, role: :admin
+
+      login_as user
+      visit search_project_portifoliorrr_profiles_path project
+
+      expect(current_path).to eq root_path
+      expect(page).to have_content 'Você não tem acesso a esse recurso'
+    end
+
+    it 'para ver o botão' do
+      project = create :project
+      user = create :user, email: 'user@email.com', cpf: '149.759.780-32'
+      create :user_role, user:, project:, role: :admin
+
+      login_as user
+      visit project_path project
+
+      expect(page).not_to have_link 'Procurar usuários'
+    end
+  end
+
+  it 'e não há colaboradores a serem exibidos' do
+    user = create :user, email: 'user@email.com', cpf: '149.759.780-32'
```

Comment:
Turma, temos um ponto aqui. 

Seria ideal a gente separar os testes de model e de integração da seguinte forma:

- Os testes do model PortfoliorrrProfile fazem o mock das chamadas ao Faraday. Como foi feito aqui no teste de sistema.
- Os testes de sistema, como este aqui em que estou comentando, fazem mock dos métodos dos modelos. Ex: 'allow(PortfoliorrrProfile).to receive(:all).and_return([])` . O retorno aqui vai ser direto um array dos objetos já certinho.

Reply:
Fizemos a separação conforme sugerido e alteramos a organização dos testes.